### PR TITLE
JW-878 utils msi

### PR DESCRIPTION
**Pull request by mgreg0r**

### CODE REVIEW

File: utils/SConscript [hunk 1]
```diff
@@ -68,6 +68,8 @@ dropstats = env.Program(target = 'dropstats', source = dropstats_sources)
 # to make sure that all are built when you do 'scons' @ the top level
 binaries  = [vif, rt, nh, mpls, flow, dropstats]
 
+
+
```
Comment (michal-clapinski) [inline utils/SConscript]: Don't make useless changes

File: utils/vtest/SConscript [hunk 2]
```diff
@@ -72,8 +72,8 @@ if not sys.platform.startswith('win'):
     env.Command('libjnprvnet.a', None, 'make', chdir=src_root+'/vrouter/utils/vtest/vhost')
 
 vtest = env.Program('vtest', vtest_src)
+Export('vtest')
 env.Requires(vtest, sandesh_gen_output)
-
```
Comment (michal-clapinski) [inline utils/vtest/SConscript]: no useless changes

File: utils/windows/utils_msi.wxs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        <Package InstallerVersion="200" Compressed="yes" Comments="Windows Installer Package" Platform="x64"/>
+        <Media Id="1" Cabinet="product.cab" EmbedCab="yes"/>
+ 
+        <Directory Id="TARGETDIR" Name="SourceDir">
+            <Directory Id="ProgramFiles64Folder">
+                <Directory Id="ManufacturerFolder" Name="Juniper Networks">
+                    <Directory Id="INSTALLDIR" Name="vRouter utilities">
+                        <Component Id="DropstatsFiles" Guid="CB895D26-EC64-498E-8FA4-E578A3C6EA49" Win64="yes">
+                            <File Id="DropstatsExe" Source="../dropstats.exe"/>
+                            <ServiceInstall Id="CreateDropstats" Name="dropstats" DisplayName="dropstats" Type="shareProcess" Start="auto" ErrorControl="normal"/>
```
Comment (michal-clapinski) [inline utils/windows/utils_msi.wxs]: utils are not services

File: utils/windows/utils_msi.wxs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                            <ServiceInstall Id="CreateVif" Name="vif" DisplayName="vif" Type="shareProcess" Start="auto" ErrorControl="normal"/>
+                            <ServiceControl Id="DeleteVif" Name="vif" Remove="uninstall"/>
+                        </Component>
+                        <Component Id="VtestFiles" Guid="13BFC527-4B1B-44F4-80E7-975FA16EA99A" Win64="yes">
+                            <File Id="VtestExe" Source="../vtest/vtest.exe"/>
+                            <ServiceInstall Id="CreateVtest" Name="vtest" DisplayName="vtest" Type="shareProcess" Start="auto" ErrorControl="normal"/>
+                            <ServiceControl Id="DeleteVtest" Name="vtest" Remove="uninstall"/>
+                        </Component>
+                        <Component Id="VrouterUtilsDLL" Guid="611FF57B-D728-4F23-8D2A-7F80509A9606" Win64="yes">
+                            <File Id="Libxml" Source="../../../../../third_party/libxml2-2.9.4/win32/bin.msvc/libxml2.dll"/>
```
Comment (michal-clapinski) [inline utils/windows/utils_msi.wxs]: The path should be something like 'build/bin/libxml2.dll'

File: .gitignore [hunk 5]
```diff
@@ -21,19 +21,19 @@ vrouter.mod.c
...
 /ipch/
 /sandesh/gen-c/*
-/utils/*obj
 /test/cmocka
 /test/atomic-test/x64
-/utils/vtest/x64
 /windows/installer/*/x64
 /windows/installer/*/Debug
 /windows/installer/*/Release
+
```
Comment (michal-clapinski) [inline .gitignore]: revert this

File: utils/windows/utils_msi.wxs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+            <ComponentRef Id="MplsFiles"/>
+            <ComponentRef Id="NhFiles"/>
+            <ComponentRef Id="RtFiles"/>
+            <ComponentRef Id="VifFiles"/>
+            <ComponentRef Id="VtestFiles"/>
+            <ComponentRef Id="VRouterUtilsPath"/>
+            <ComponentRef Id="VrouterUtilsDLL"/>
+        </Feature>
+    </Product>
+</Wix>
```
Comment (michal-clapinski) [inline utils/windows/utils_msi.wxs]: Add newline

**Comment by mgreg0r** [comment]
Retest this please

File: utils/SConscript [hunk 7]
```diff
@@ -16,7 +16,14 @@ if system_header_path:
     env.Append(CPPPATH = system_header_path + '/include/')
 
 # CFLAGS
-env.Append(CCFLAGS = '-g')
+if sys.platform.startswith('win'):
+    env.Append(CPPDEFINES='_WINDOWS')
+    #env.Append(CCFLAGS = '/MTd')
```
Comment (m-kostrzewa) [inline utils/SConscript]: Why is this commented out? (same in utils/vtest/SConscript file)

**Pull request merged**